src/commands/deploy/deploy.ts

```diff
@@ -45,7 +45,13 @@ async function deploy(actionContext: IActionContext, arg1: vscode.Uri | string |
 
     addLocalFuncTelemetry(actionContext, deployPaths.workspaceFolder.uri.fsPath);
 
-    const projectPath: string = nonNullValue(await tryGetFunctionProjectRoot(actionContext, deployPaths.workspaceFolder));
+    const projectPath: string | undefined = await tryGetFunctionProjectRoot(actionContext, deployPaths.workspaceFolder);
+    if (projectPath === undefined) {
+        const message: string = localize('functionProjectRootNotFound', 'No azure function project root could be found. This can be caused by a missing {0} file.', hostFileName);
+        await actionContext.ui.showWarningMessage(message, { stepName: "functionProjectRootNotFound" })
+        return
```

Comment:
```suggestion
        throw new Error(message);
```

Throwing an exception here would be best as it stops execution, and will automatically show an error notification.